torch/csrc/jit/script/script_type_parser.h

```diff
@@ -28,6 +28,8 @@ class TORCH_API ScriptTypeParser {
 
   FunctionSchema parseSchemaFromDef(const Def& def, bool skip_self);
 
+  c10::IValue parseConstant(const Expr& type, const Expr& expr);
```

Comment:
This seems the the wrong place to put this code. First, it doesn't use `this` and this is the type parser, so it should not expose an API that parses values. Maybe make it a separate function to make this clear.

Reply:
It does use `parseBaseTypeName` and `evaluateDefaults` which are part of `ScriptTypeParser` and constant is part of type as of now..maybe I can change the name to `parseClassConstant` to make it more clear?

Reply:
> Generally looks good to me with a small restructuring. I'd like to see tests that verify that we can save/load with constant information intact, but they probably need to be added later in the stack, when you've exposed constant stuff to `Module`/Python.

Yeah, we can add official tests later, I have test locally to make sure this works as well.